cli/trust/trust.go

```diff
@@ -299,9 +299,12 @@ type ImageRefAndAuth struct {
 	digest     digest.Digest
 }
 
+// RepositoryInfoResolver returns repository info for the given reference.
+type RepositoryInfoResolver func(name reference.Named) (*registry.RepositoryInfo, error)
```

Comment:
I was a bit on the fence wether to define this as a type, or to just inline the signature in the function below (like we do for `authResolver`).